src/olympia/api/paginator.py

```diff
@@ -51,13 +55,19 @@ def page(self, number):
         # Force the search to evaluate and then attach the count. We want to
         # avoid an extra useless query even if there are no results, so we
         # directly fetch the count from hits.
-        # Overwrite `object_list` with the list of ES results.
-
-        result = self.object_list[bottom:top].execute()
-
-        page = Page(result.hits, number, self)
-        # Update the `_count`.
-        self._count = page.object_list.total
+        if self.force_legacy_compat:
```

Comment:
I'm not a huge fan of calling the parameter `force_legacy_compat`, but I don't have better suggestions at the moment. In any case, add a comment here and a docstring in `__init__` to explain what the parameter is - something that would talk about elasticsearch-dsl vs our own `ES` layer